server/src/test/java/org/elasticsearch/action/admin/indices/diskusage/IndexDiskUsageAnalyzerTests.java

```diff
@@ -0,0 +1,526 @@
+            .setUseCompoundFile(randomBoolean())
+            .setCodec(new Lucene87Codec(codecMode.mode()));
+        try (IndexWriter writer = new IndexWriter(directory, config)) {
+            for (int i = 0; i < numDocs; i++) {
+                final Document doc = new Document();
+                addFields.accept(doc);
+                writer.addDocument(doc);
+            }
+        }
+    }
+
+    static void addRandomDocValuesField(Document doc) {
+        if (randomBoolean()) {
+            doc.add(new NumericDocValuesField("ndv", random().nextInt(1024)));
+        }
+        if (randomBoolean()) {
+            doc.add(new BinaryDocValuesField("bdv", new BytesRef(randomAlphaOfLength(3))));
+        }
+        if (randomBoolean()) {
+            doc.add(new SortedDocValuesField("sdv", new BytesRef(randomAlphaOfLength(3))));
+        }
+        int numValues = random().nextInt(5);
+        for (int i = 0; i < numValues; ++i) {
+            doc.add(new SortedSetDocValuesField("ssdv", new BytesRef(randomAlphaOfLength(3))));
+        }
+        numValues = random().nextInt(5);
+        for (int i = 0; i < numValues; ++i) {
+            doc.add(new SortedNumericDocValuesField("sndv", random().nextInt(1024)));
+        }
+    }
+
+    static void addRandomPostings(Document doc) {
+        for (IndexOptions opts : IndexOptions.values()) {
+            if (opts == IndexOptions.NONE) {
+                continue;
+            }
+            FieldType ft = new FieldType();
+            ft.setIndexOptions(opts);
+            ft.freeze();
+            final int numFields = random().nextInt(5);
+            for (int j = 0; j < numFields; ++j) {
+                doc.add(new Field("f_" + opts, randomAlphaOfLength(5), ft));
+            }
+        }
+    }
+
+    static void addRandomStoredFields(Document doc, int numFields) {
+        final int numValues = random().nextInt(3);
+        for (int i = 0; i < numValues; ++i) {
+            final String field = "sf-" + between(1, numFields);
+            if (randomBoolean()) {
+                doc.add(new StoredField(field, randomAlphaOfLength(5)));
+            } else {
+                doc.add(new StoredField(field, randomLong()));
+            }
+        }
+    }
+
+    static void addRandomPoints(Document doc) {
+        final int numValues = random().nextInt(5);
+        for (int i = 0; i < numValues; i++) {
+            doc.add(new IntPoint("pt-" + randomIntBetween(1, 2), random().nextInt()));
+        }
+    }
+
+    static FieldType randomVectorFieldType() {
+        FieldType fieldType = new FieldType(TextField.TYPE_NOT_STORED);
+        fieldType.setStoreTermVectors(true);
+        fieldType.setStoreTermVectorPositions(randomBoolean());
+        fieldType.setStoreTermVectorOffsets(randomBoolean());
+        // TODO: add payloads
+        fieldType.setStoreTermVectorPayloads(false);
+        return fieldType;
+    }
+
+    static void addRandomVectors(Document doc) {
```

Comment:
Rename Vectors to TermVectors to prepare for the introduction of vectors in Lucene?